gnucash/gnome/gnc-plugin-page-register.c

```diff
@@ -5019,6 +5028,8 @@ gnc_plugin_page_register_cmd_scrub_all (GtkAction* action,
             g_free (progress_msg);
         }
 
+        if (abort_scrub) break;
+
```

Comment:
Why?

Reply:
Why move the `abort_scrub` test from the for loop to here? ISTM that `gnc_window_show_progress` which calls `gtk_main_iteration` seemed to modify something which results in the `split` object being invalid, therefore aborting the scrub led to a `g_warning`. I'll check again.

Reply:
Confirmed. While checking `node && !abort_scrub` is more elegant, it means: if a lengthy scrub is running, and user closes via Alt-F4 (on Linux), and agrees to abort, the `split` becomes `NULL` which means `scrub_split(split)` fails.

Reply:
So move `scrub_split` and its `PINFO` to before the progress bar update.